backend.native/compiler/ir/backend.native/src/org/jetbrains/kotlin/backend/konan/ObjCInterop.kt

```diff
@@ -80,6 +81,7 @@ fun IrClass.isObjCProtocolClass(): Boolean =
 fun ClassDescriptor.isObjCProtocolClass(): Boolean =
         this.fqNameSafe == objCProtocolFqName
 
+@Suppress("DEPRECATION")
```

Comment:
Can it be properly fixed?

Reply:
This method uses locally declared deprecated function. It'd be better to drop suppression because it's already covered by file:Suppress

Reply:
Usually file-level suppresses are not very good idea.

Reply:
Btw, as far as I know there are a lot of descriptor-based functions **deprecated by mistake**: they are used in the code that is to be migrated to frontend checkers.

Reply:
Shall we remove deprecations then?

Reply:
Yes. Generally we should remove deprecations from methods that are directly or indirectly used to decide about `reportCompilationError` usage.